src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -271,6 +271,21 @@ class ResourceService(
         }
       )
 
+  def setResourceAuthDomain(
+      resource: FullyQualifiedResourceId,
+      authDomains: Set[WorkbenchGroupName],
+      userId: WorkbenchUserId,
+      samRequestContext: SamRequestContext
+  ): IO[Set[WorkbenchGroupName]] =
+    for {
+      resourceType <- getResourceType(resource.resourceTypeName)
+      policies <- listResourcePolicies(resource, samRequestContext)
+      _ <- validateAuthDomain(resourceType.get, authDomains, userId, samRequestContext)
+      _ <- cloudExtensions.onGroupUpdate(policies.map(p => FullyQualifiedPolicyId(resource, p.policyName)), samRequestContext)
```

Comment:
`onGroupUpdate` should come after `accessPolicyDAO.setResourceAuthDomain` to avoid a race condition where the message is handled before the database is updated

Reply:
Thanks Doug, I went back and forth on the order there. This makes sense.